app/src/behaviors/behavior_hold_tap.c

```diff
@@ -140,14 +141,13 @@ static void release_captured_events() {
         if (undecided_hold_tap != NULL) {
             k_msleep(10);
         }
-        if (is_zmk_position_state_changed(captured_event)) {
-            struct zmk_position_state_changed *position_event =
-                cast_zmk_position_state_changed(captured_event);
+
+        struct zmk_position_state_changed *position_event;
+        struct zmk_keycode_state_changed *modifier_event;
+        if ((position_event = as_zmk_position_state_changed(captured_event))) {
```

Comment:
Up to you what you think is easier to read, but in the code I've seen at work, we usually silence the assignment inside "if" warning with

```c
if ((ptr = expression()) != NULL)
```

just to make it super clear that yes, I did mean this as an assignment, and here's a definitely Boolean expression to prove it.

Reply:
I'm fine w/ moving to the more verbose version. Updated.